src/framework/xr/xr-mesh.js

```diff
@@ -0,0 +1,179 @@
+import { EventHandler } from "../../core/event-handler.js";
+import { Vec3 } from "../../core/math/vec3.js";
+import { Quat } from "../../core/math/quat.js";
+
+let ids = 0;
+
+/**
+ * Detected Mesh instance that provides its transform (position, rotation),
+ * triangles (vertices, indices) and its semantic label. Any of its properties can
+ * change during its lifetime.
+ *
+ * @category XR
+ */
+class XrMesh extends EventHandler {
+    /**
+     * @type {number}
+     * @private
+     */
+    _id;
+
+    /**
+     * @type {import('./xr-mesh-detection.js').XrMeshDetection}
+     * @private
+     */
+    _meshes;
```

Comment:
this could be perhaps better called `meshDetection` .. meshes implies it's an array of some meshes